src/librustc_trans/debuginfo/mod.rs

```diff
@@ -168,7 +168,7 @@ pub fn finalize(cx: &CrateContext) {
     }
 
     debug!("finalize");
-    let _ = compile_unit_metadata(cx);
+    //let _ = compile_unit_metadata(cx);
```

Comment:
Although thinking of what @alexcrichton was saying, it might have to just be placed in an `#if` guard to maintain compatibility with earlier LLVM...